provider/datagen/src/transform/icuexport/collator/test.rs

```diff
@@ -1,312 +0,0 @@
-    let mut options = CollatorOptions::new();
-
-    options.strength = Some(Strength::Tertiary);
-    check_expectations(&locale, options, &cases);
-
-    options.strength = Some(Strength::Primary);
-    check_expectations(&locale, options, &cases);
-}
-
-#[test]
-fn test_nb_nn_no() {
-    let provider = crate::DatagenProvider::for_test();
-
-    let input = vec!["ü", "y", "å", "ø"];
-    let expected = &["y", "ü", "ø", "å"];
-
-    // Test "no" macro language without fallback (should equal expected)
-    let collator =
-        Collator::try_new_unstable(&provider, &locale!("no").into(), CollatorOptions::new())
-            .unwrap();
-    let mut strs = input.clone();
-    strs.sort_by(|a, b| collator.compare(a, b));
-    assert_eq!(strs, expected);
-
-    // Test "und" without fallback (should NOT equal expected)
-    let collator =
-        Collator::try_new_unstable(&provider, &locale!("und").into(), CollatorOptions::new())
-            .unwrap();
-    let mut strs = input.clone();
-    strs.sort_by(|a, b| collator.compare(a, b));
-    assert_ne!(strs, expected);
-
-    // Test "nb" without fallback (should fail to load)
-    if Collator::try_new_unstable(&provider, &locale!("nb").into(), CollatorOptions::new()).is_ok()
-    {
-        panic!("Should fail to create 'nb' without fallback enabled")
-    }
-
-    // Enable locale fallback on the provider now
-    let provider = LocaleFallbackProvider::try_new_unstable(provider).unwrap();
-    let provider = ResolvedLocaleAdapter::new(provider);
-
-    // Test "no" macro language WITH fallback (should equal expected)
-    let input_locale = locale!("no").into();
-    let collator =
-        Collator::try_new_unstable(&provider, &input_locale, CollatorOptions::new()).unwrap();
-    let mut strs = input.clone();
-    strs.sort_by(|a, b| collator.compare(a, b));
-    assert_eq!(strs, expected);
-    assert_eq!(
-        provider.resolved_locale_for(CollationDataV1Marker::KEY, input_locale),
-        Some(&locale!("no").into())
-    );
-
-    // Now "nb" should work
-    let input_locale = locale!("nb").into();
-    let collator =
-        Collator::try_new_unstable(&provider, &input_locale, CollatorOptions::new()).unwrap();
-    let mut strs = input.clone();
-    strs.sort_by(|a, b| collator.compare(a, b));
-    assert_eq!(strs, expected);
-    assert_eq!(
-        provider.resolved_locale_for(CollationDataV1Marker::KEY, input_locale),
-        Some(&locale!("no").into())
-    );
-
-    // And "nn" should work, too
-    let input_locale = locale!("nn").into();
-    let collator =
-        Collator::try_new_unstable(&provider, &input_locale, CollatorOptions::new()).unwrap();
-    let mut strs = input.clone();
-    strs.sort_by(|a, b| collator.compare(a, b));
-    assert_eq!(strs, expected);
-    assert_eq!(
-        provider.resolved_locale_for(CollationDataV1Marker::KEY, input_locale),
-        Some(&locale!("no").into())
-    );
-}
-
-#[test]
-fn test_zh() {
```

Comment:
Issue: You lost this test

```rust
#[test]
fn test_zh() {
    let mut provider = crate::DatagenProvider::for_test();
    // `zh-u-co-gb2312` needs to be manually enabled
    provider.source.options.collations.insert("gb2312".into());
    let provider = LocaleFallbackProvider::try_new_unstable(provider).unwrap();
    let provider = ResolvedLocaleAdapter::new(provider);

    let input_locale = locale!("zh-u-co-gb2312").into();
    Collator::try_new_unstable(&provider, &input_locale, CollatorOptions::new()).unwrap();
    assert_eq!(
        provider.resolved_locale_for(CollationDataV1Marker::KEY, input_locale),
        Some(&locale!("zh-u-co-gb2312").into())
    );
}
```

Reply:
As this just tests construction I moved this to the collations selection test

Reply:
It tests that you can add `gb2312` to datagen and then load `gb2312` and it works.

Reply:
It is supposed to that gb2312 actually "works" rather than just showing up in the collation filter like the other test does.

Reply:
The new test asserts that the `zh-u-co-gb2312` locale gets put into the `DataExporter`, so clearly it's available for loading?

Reply:
The test tests that it shows up in `IterableDataProvider::<CollationDataV1Marker>::supported_locales(&provider)` but that's a slightly weaker assertion than asserting that it actually makes it all the way through to the Collator constructor